pallets/sto/src/lib.rs

```diff
@@ -471,48 +524,48 @@ impl<T: Trait> Module<T> {
         fundraiser_id: u64,
         frozen: bool,
     ) -> DispatchResult {
-        Self::ensure_perms_pia(origin, &offering_asset)?;
+        let did = Self::ensure_perms_pia(origin, &offering_asset)?.0;
+        let mut fundraiser = <Fundraisers<T>>::get(offering_asset, fundraiser_id)
+            .ok_or(Error::<T>::FundraiserNotFound)?;
         ensure!(
-            <Fundraisers<T>>::contains_key(offering_asset, fundraiser_id),
-            Error::<T>::FundraiserNotFound
+            fundraiser.status != FundraiserStatus::Closed,
+            Error::<T>::FundraiserClosed
         );
-        <Fundraisers<T>>::mutate(offering_asset, fundraiser_id, |fundraiser| {
-            if let Some(fundraiser) = fundraiser {
-                fundraiser.frozen = frozen
-            }
-        });
+        if frozen {
+            fundraiser.status = FundraiserStatus::Frozen;
+            Self::deposit_event(RawEvent::FundraiserFrozen(did, fundraiser_id));
+        } else {
+            fundraiser.status = FundraiserStatus::Live;
+            Self::deposit_event(RawEvent::FundraiserUnfrozen(did, fundraiser_id));
+        }
```

Comment:
```suggestion
        let (status, event) = if frozen {
            (FundraiserStatus::Frozen, RawEvent::FundraiserFrozen)
        } else {
            (FundraiserStatus::Live, RawEvent::FundraiserUnfrozen)
        };
        fundraiser.status = status;
        Self::deposit_event(event(did, fundraiser_id));
```

Reply:
This change breaks the build. I don't think it's worth the hassle so just reverting this.

Reply:
https://app.circleci.com/pipelines/github/PolymathNetwork/Polymesh/6872/workflows/ab1d4ea6-111d-40ab-8759-f743264759da/jobs/18741